pkg/internal/vault/vault.go

```diff
@@ -258,6 +259,13 @@ func (v *Vault) requestTokenWithAppRoleRef(client Client, appRole *v1.VaultAppRo
 		return "", fmt.Errorf("error encoding Vault parameters: %s", err.Error())
 	}
 
+	vaultIssuer := v.issuer.GetSpec().Vault
+	if vaultIssuer.Namespace != "" {
+		vaultReqHeaders := http.Header{}
+		vaultReqHeaders.Add("X-VAULT-NAMESPACE", vaultIssuer.Namespace)
+		request.Headers = vaultReqHeaders
```

Comment:
this overrides all request headers, we should avoid doig that

Reply:
Yeah. Good point. I see we have done something similar [here](https://github.com/jetstack/cert-manager/blob/master/pkg/internal/vault/vault.go#L112-L116) as well. Let me go ahead and fix both the places.